sdks/python/apache_beam/typehints/native_type_compatibility.py

```diff
@@ -84,19 +84,38 @@ def _safe_issubclass(derived, parent):
     issubclass(derived, parent), or False if a TypeError was raised.
   """
   try:
-    return issubclass(derived, parent)
+    if type(derived) == typing._GenericAlias:
+      out = issubclass(derived.__origin__, parent.__origin__)
+    else:
+      out = issubclass(derived, parent)
+    return out
   except TypeError:
     return False
+  except AttributeError:
+    try:
+      out = issubclass(derived, parent)
+      return out
+    except TypeError:
+      return False
 
 
 def _match_issubclass(match_against):
   return lambda user_type: _safe_issubclass(user_type, match_against)
 
 
+def _matcher_(derived, parent):
```

Comment:
Let's make it a nested function inside _match_same_type:

```
def _match_same_type(match_against):
  def matcher(user_type):
      # Compare user_type and match_against here
  
  return lambda user_type: match(user_type)
```

Reply:
Sounds good!